source/common/stats/histogram_impl.cc

```diff
@@ -79,5 +79,64 @@ void HistogramStatisticsImpl::refresh(const histogram_t* new_histogram_ptr) {
   }
 }
 
+ParentHistogramImpl::ParentHistogramImpl(const std::string& name, Store& parent,
+                                         std::string&& tag_extracted_name, std::vector<Tag>&& tags)
+    : MetricImpl(std::move(tag_extracted_name), std::move(tags)), parent_(parent),
+      interval_histogram_(hist_alloc()), cumulative_histogram_(hist_alloc()),
```

Comment:
You can also use std::unique_ptr<> with a custom deleter, so you can get rid of the dtor.

Reply:
Yeah, there's a `CSmartPtr` helper for this very thing. I didn't use it here just to stay consistent with `ThreadLocalParentHistogram`, which also doesn't use it, but perhaps they both should.

Reply:
Actually, decided not to change this. Too many `.get()`s.